### PR TITLE
Feat: exposes context on mixed.test function and add originalValue to context

### PR DESCRIPTION
**Pull request by abnersajr**
This PR fixes two issues:  #207 and #1007.
The discussions suggest having `originalValue` exposed in some way. Exposing as the second argument was one of the suggestions.
Following the API and suggestion from @jquense, would be better to have this `originalValue` into the  `this` context.

In previous versions, you can't use arrow functions too, because it's being used `.call(ctx...)`.
With these changes, you can use both ways. The `this` context is now exposed too as a second argument.
Eg: `mixed.test('name', 'test name', (value, ctx) => ctx.originalValue)`

Also edited Type Definitios: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/47069

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -706,26 +706,27 @@ use the alternate signature to provide more options (see below):
 let jimmySchema = string().test(
   'is-jimmy',
   '${path} is not Jimmy',
-  value => value === 'jimmy',
+  value, context => value === 'jimmy',
```
Comment (jquense) [inline README.md:709]:
```suggestion
  (value, context) => value === 'jimmy',
```

**Comment by jquense** [comment]
Hi @abnersajr thanks for the PR. It looks like this PR updates the docs but doesn't implement the feature. Is that intentional or did a commit go missing somewhere?

**Comment by abnersajr** [comment]
Sorry @jquense. Definitely something was misssed in my commits. I'm gonna check it ASAP. and thanks for the quick review.

**Comment by alexandernst** [comment]
What happened to this?

**Comment by alexandernst** [comment]
@jquense Are you're still willing to merge this? If there is something missing, I can fork this branch and add it.

**Comment by alexandernst** [comment]
@jquense ping

**Comment by alexandernst** [comment]
@jquense ping

**Comment by alexandernst** [comment]
@jquense ping

**Comment by jquense** [comment]
can you resolve the conflict @abnersajr ?

**Comment by abnersajr** [comment]
@jquense sure. I'm gonna do this.

**Comment by abnersajr** [comment]
@jquense merge is done. Solved the conflicts adding to the sync and async version.

**Comment by jquense** [comment]
thanks!

**Comment by alexandernst** [comment]
\o/

can we get a new minor release with this, please? 😄

**Comment by alexandernst** [comment]
@jquense Actually, I think this might require to bump the version up to `0.30.0` instead of `0.29.4`.

**Pull request merged**